pkg/cloud/cloud.go

```diff
@@ -1604,6 +1616,28 @@ func (c *cloud) EnableFastSnapshotRestores(ctx context.Context, availabilityZone
 	return response, nil
 }
 
+// DryRun will make a dry-run EC2 API call. Nil return value means we successfully received EC2 DryRunOperation error code.
+func (c *cloud) DryRun(ctx context.Context) error {
+	if c.attemptDryRun.Load() {
+		_, apiErr := c.ec2.DescribeAvailabilityZones(ctx,
+			&ec2.DescribeAvailabilityZonesInput{DryRun: aws.Bool(true)},
+			func(o *ec2.Options) {
+				o.Retryer = aws.NopRetryer{} // Don't retry so we can catch network failures. CO should retry liveness check multiple times.
+				o.APIOptions = nil           // Don't add our logging/metrics middleware because we expect errors.
+			})
+		if apiErr != nil {
+			var awsErr smithy.APIError
+			if errors.As(apiErr, &awsErr) && awsErr.ErrorCode() == "DryRunOperation" {
+				c.attemptDryRun.Store(false)
+				return nil
+			}
+			return fmt.Errorf("dry-run EC2 API call failed, EC2 API err: %w", apiErr)
```

Comment:
Suggest changing string to `dry-run EC2 API call failed: %w`, as the error may not be from the EC2 API (could be a networking/DNS issue for example, or from STS). This also cuts a few characters off an otherwise long error message.